typing_inspect/test_typing_inspect.py

```diff
@@ -0,0 +1,124 @@
+from typing_inspect import (
+    is_generic_type, is_callable_type, is_tuple_type, is_union_type,
+    is_typevar, is_classvar, get_origin, get_parameters, get_last_args, get_args,
+    get_generic_type, get_generic_bases,
+)
+from unittest import TestCase, main
+from typing import (
+    Union, ClassVar, Callable, Optional, TypeVar, Sequence, Mapping,
+    MutableMapping, Iterable, Generic, List, Any, Dict, Tuple,
+)
+
+
+class IsUtilityTestCase(TestCase):
+    def sample_test(self, fun, samples, nonsamples):
+        for s in samples:
+            self.assertTrue(fun(s))
+        for s in nonsamples:
+            self.assertFalse(fun(s))
+
+    def test_generic(self):
+        T = TypeVar('T')
+        samples = [Generic, Generic[T], Iterable[int], Mapping,
+                   MutableMapping[T, List[int]], Sequence[Union[str, bytes]]]
+        nonsamples = [int, Union[int, str], Union[int, T], ClassVar[List[int]],
+                      Callable[..., T], ClassVar, Optional, bytes, list]
+        self.sample_test(is_generic_type, samples, nonsamples)
+
+    def test_callable(self):
+        samples = [Callable, Callable[..., int],
+                   Callable[[int, int], Iterable[str]]]
+        nonsamples = [int, type, 42, [], List[int],
+                      Union[callable, Callable[..., int]]]
+        self.sample_test(is_callable_type, samples, nonsamples)
+        class MyClass(Callable[[int], int]):
+            pass
+        self.assertTrue(is_callable_type(MyClass))
+
+    def test_tuple(self):
+        samples = [Tuple, Tuple[str, int], Tuple[Iterable, ...]]
+        nonsamples = [int, tuple, 42, List[int]]
```

Comment:
Maybe add some NamedTuples as test cases; I assume they'll be nonsamples because typing.Tuple and typing.NamedTuple are completely distinct, but it would be good to clarify anyway.